src/backend/gporca/libgpos/src/error/CFSimulator.cpp

```diff
@@ -131,18 +131,13 @@ CFSimulator::NewStack(ULONG major, ULONG minor)
 //		Initialize global instance
 //
 //---------------------------------------------------------------------------
-GPOS_RESULT
+void
 CFSimulator::Init()
 {
-	CAutoMemoryPool amp;
-	CMemoryPool *mp = amp.Pmp();
+	CMemoryPool *mp =
+		CMemoryPoolManager::GetMemoryPoolMgr()->CreateMemoryPool();
 
 	CFSimulator::m_fsim = GPOS_NEW(mp) CFSimulator(mp, GPOS_FSIM_RESOLUTION);
```

Comment:
Doesn't we need to wrap it into try/catch, like it's done in other `Init` methods and free the `mp` at catch, if error occurred at stage of creation of CFSimulator?

Reply:
> if error occurred at stage of creation of CFSimulator

then CFSimulator::Shutdown method is called and it frees mp

Reply:
but what if Shutdown won't be called for some reasons (for ex. Init will be used in other part of code, but Shutdown won't be by mistake)?
As I said before, seems it'll be nice to unify approaches. Other Init methods do the things with try/catch, and it for me it seems safe or it's a redundant work?

Reply:
> Other Init methods do the things with try/catch

They have a condition in shutdown methods for destroy mp, but CFSimulator has not!
If I destroy mp in CFSimulator::Init, how I destroy it in CFSimulator::Shutdown? Nullify mp and then check mp for NULL?

Reply:
Sorry, i thought that Shutdown of CFSimulator is static function, but it's not. Ok what would be if the exception would occur on this instruction? As I understand `m_fsim` would be `NULL`, TerminateGPOPT would call gpos_terminate and here is a check:
```
	if (NULL != CFSimulator::FSim())
	{
		CFSimulator::FSim()->Shutdown();
	}
```
So possible leak
Am I wrong?

Reply:
Yes, you are right, I corrected it.

Reply:
Please, verify other places where AutoMemoryPool was changed

Reply:
done